src/flows/pipes/Notification/HTTP.tsx

```diff
@@ -0,0 +1,138 @@
+import React, {FC, useContext} from 'react'
+import {
+  Form,
+  FlexBox,
+  Input,
+  InputType,
+  ComponentSize,
+  SelectGroup,
+} from '@influxdata/clockface'
+
+import {PipeContext} from 'src/flows/context/pipe'
+
+const HTTP: FC = () => {
+  const {data, update} = useContext(PipeContext)
+
+  const updater = (field, value) => {
+    update({
+      endpointData: {
+        ...data.endpointData,
+        [field]: value,
+      },
+    })
+  }
+
+  const updateUrl = evt => {
+    updater('url', evt.target.value)
+  }
+
+  const updateAuth = (auth: string) => {
+    updater('auth', auth)
+  }
+
+  const updateUsername = evt => {
+    updater('username', evt.target.value)
+  }
+
+  const updatePassword = evt => {
+    updater('password', evt.target.value)
+  }
+
+  const updateToken = evt => {
+    updater('token', evt.target.value)
+  }
+
+  let submenu
+
+  if (data.endpointData.auth === 'basic') {
+    submenu = (
+      <>
+        <Form.Element label="Username">
+          <Input
+            name="username"
+            type={InputType.Text}
+            value={data.endpointData.username}
+            onChange={updateUsername}
+            size={ComponentSize.Medium}
+          />
+        </Form.Element>
+        <Form.Element label="Password">
+          <Input
+            name="password"
+            type={InputType.Text}
+            value={data.endpointData.password}
+            onChange={updatePassword}
+            size={ComponentSize.Medium}
+          />
+        </Form.Element>
+      </>
+    )
+  } else if (data.endpointData.auth === 'bearer') {
+    submenu = (
+      <Form.Element label="Token">
+        <Input
+          name="token"
+          type={InputType.Text}
+          value={data.endpointData.token}
+          onChange={updateToken}
+          size={ComponentSize.Medium}
+        />
+      </Form.Element>
+    )
```

Comment:
won't every panel make a copy of this? Is it worth it to move these into their own functional component?

Reply:
for the sake of.. having more components in a folder? i dont understand what is making copies here